### PR TITLE
[Integ Test] Fix test_build_image by adding required export logs policy to CustomS3Bucket

### PR DESCRIPTION
**Pull request by hehe7318**
### Description of changes
* Fix test_build_image by adding required export logs policy to CustomS3Bucket

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hanwen-pcluste — APPROVED** [review]

File: tests/integration-tests/tests/createami/test_createami.py [hunk 1]
```diff
@@ -319,32 +320,36 @@ def _test_get_image_log_events(image):
...
 
+def _set_s3_bucket_policy(bucket_name, partition, region):
+    bucket_policy = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Action": "s3:GetBucketAcl",
+                "Effect": "Allow",
+                "Resource": f"arn:{partition}:s3:::{bucket_name}",
+                "Principal": {"Service": f"logs.{region}.amazonaws.com"},
```
Comment (gmarciani) [inline tests/integration-tests/tests/createami/test_createami.py:331]: Shouldn't the AWS domain be adapted according to the partition?

File: tests/integration-tests/tests/createami/test_createami.py [hunk 2]
```diff
@@ -319,32 +320,36 @@ def _test_get_image_log_events(image):
...
+            },
+            {
+                "Action": "s3:PutObject",
+                "Effect": "Allow",
+                "Resource": f"arn:{partition}:s3:::{bucket_name}/*",
+                "Condition": {"StringEquals": {"s3:x-amz-acl": "bucket-owner-full-control"}},
+                "Principal": {"Service": f"logs.{region}.amazonaws.com"},
```
Comment (gmarciani) [inline tests/integration-tests/tests/createami/test_createami.py:338]: Same as above

**Review by gmarciani — CHANGES_REQUESTED** [review]

File: tests/integration-tests/tests/createami/test_createami.py (hunk 1 above)
Comment (hehe7318) [reply tests/integration-tests/tests/createami/test_createami.py:331]: That's right, so how can it work previously

Comment (gmarciani) [reply tests/integration-tests/tests/createami/test_createami.py:331]: Good point, apparently in these partitions the commercial domain suffix works. I expect the same in US ISO region at this point. We are good to go.

**Review by gmarciani — APPROVED** [review]

**Pull request merged**